src/providers/MeetingProvider/MeetingManager.ts

```diff
@@ -215,29 +167,100 @@ export class MeetingManager implements AudioVideoObserver {
+    const {
+      deviceLabels,
+      eventController,
+      logger,
+      enableWebAudio,
+      activeSpeakerPolicy,
+    } = this.parseJoinParams(meetingSessionConfiguration, options);
+    this.meetingSessionConfiguration = meetingSessionConfiguration;
+    this.meetingId = this.meetingSessionConfiguration.meetingId;
+    this.logger = logger;
+
+    const deviceController = new DefaultDeviceController(logger, {
+      enableWebAudio: enableWebAudio,
+    });
+    this.meetingSession = new DefaultMeetingSession(
+      meetingSessionConfiguration,
+      logger,
+      deviceController,
+      eventController,
     );
 
-    this.meetingRegion = meetingInfo.MediaRegion;
-    this.meetingId = this.configuration.meetingId;
+    this.audioVideo = this.meetingSession.audioVideo;
 
     if (eventController) {
-      this.eventController = eventController;
+      eventController.addObserver(this.eventDidReceiveRef);
+    } else {
+      this.meetingSession.eventController.addObserver(this.eventDidReceiveRef);
+    }
+
+    this.setupAudioVideoObservers();
+    this.setupDeviceLabelTrigger(deviceLabels);
+    await this.listAndSelectDevices(deviceLabels);
+    this.publishAudioVideo();
+    this.setupActiveSpeakerDetection(activeSpeakerPolicy);
+    this.meetingStatus = MeetingStatus.Loading;
+    this.publishMeetingStatus();
+  }
+
+  private parseJoinParams(
+    meetingSessionConfiguration: MeetingSessionConfiguration,
+    options: MeetingManagerJoinOptions | undefined
+  ) {
+    let deviceLabels: DeviceLabels | DeviceLabelTrigger;
+    let eventController: EventController | undefined;
+    let logLevel: LogLevel;
+    let postLoggerConfig: PostLoggerConfig | undefined;
+    let logger: Logger;
+    let enableWebAudio: boolean;
+    let activeSpeakerPolicy: ActiveSpeakerPolicy;
+
+    // TODO: The following if blocks can be replaced with ?? null-coalescing operator when the SDK supportes es 2020.
+    if (options?.deviceLabels) {
+      deviceLabels = options?.deviceLabels;
+    } else {
+      deviceLabels = DeviceLabels.AudioAndVideo;
+    }
+    eventController = options?.eventController;
+    if (options?.logLevel) {
+      logLevel = options.logLevel;
+    } else {
+      logLevel = LogLevel.WARN;
+    }
+    postLoggerConfig = options?.postLoggerConfig;
+    if (options?.logger) {
+      logger = options?.logger;
+    } else {
+      logger = this.createLogger(
+        logLevel,
+        postLoggerConfig,
+        meetingSessionConfiguration
+      );
+    }
+    if (options?.enableWebAudio) {
+      enableWebAudio = options?.enableWebAudio;
+    } else {
+      enableWebAudio = false;
+    }
+    if (options?.activeSpeakerPolicy) {
+      activeSpeakerPolicy = options?.activeSpeakerPolicy;
+    } else {
+      activeSpeakerPolicy = new DefaultActiveSpeakerPolicy();
     }
 
-    await this.initializeMeetingSession(this.configuration, deviceLabels);
+    return {
+      deviceLabels,
+      eventController: eventController,
```

Comment:
can we do link other variables since the key and the value is same for `eventController`?

Reply:
No I don't think we can do that cause `eventController` could be `undefined`